### PR TITLE
use dependabot for periodically updating common/

### PR DESCRIPTION
**Pull request by zmiklank**
This PR adds configuration for dependabot which will once a week check state of common/ and create PR with updated common/ if that is needed.

### CODE REVIEW

**Comment by zmiklank** [comment]
@jamacku could you please review?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+version: 1
+updates:
+  - package-ecosystem: gitsubmodule
+    directory: /
+    schedule:
+      interval: weekly
+    labels:
+      - 'merge-strategy: rebase'
+      - 'type: dependencies'
```
Comment (jamacku) [inline .github/dependabot.yml:11]: FYI: `merge-strategy: rebase` and `type: dependencies` are just labels. Not sure if you want to use them.

I have additional automation that uses these two labels, generates changelog/release notes, and automatically merges PRs.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+
+version: 1
```
Comment (jamacku) [inline .github/dependabot.yml:3]: There is already version `2`.

**Review by jamacku — COMMENTED** [review]
Looks good, just two inline notes.

Since you are using GitHub Actions, did you considered also adding:

```yml
- package-ecosystem: github-actions
    directory: /
    schedule:
      interval: monthly
    labels:
      - 'github-actions'
```

**Comment by zmiklank** [comment]
@jamacku Thank you for review. All the issues are now addressed in new commits.

**Review by jamacku — APPROVED** [review]
Looks great! :100:

**Comment by zmiklank** [comment]
@phracek When this PR is merged, the dependabot will create a PR with updated common/ submodule if that is needed. (And once a month PR with bumps of github actions if available). We talked about it in the container meeting. Are you OK with merging?

**Pull request merged**